--- sysvinit-2.96-old/src/bootlogd.c	2019-09-11 12:43:12.000000000 -0500
+++ sysvinit-2.96/src/bootlogd.c	2020-09-09 14:11:47.228553424 -0500
@@ -83,6 +83,8 @@
 	char	*dev1;
 	char	*dev2;
 } consdev[] = {
+	{ "ttymxc",	"/dev/ttymxc%s",		NULL		},
+	{ "ttyO",	"/dev/ttyO%s",		NULL		},
 	{ "ttyB",	"/dev/ttyB%s",		NULL		},
 	{ "ttySC",	"/dev/ttySC%s",		"/dev/ttsc/%s"	},
 	{ "ttyS",	"/dev/ttyS%s",		"/dev/tts/%s"	},
