diff --git a/kernel/ipipe/timer.c b/kernel/ipipe/timer.c
index 0da956413890..421eacfe098e 100644
--- a/kernel/ipipe/timer.c
+++ b/kernel/ipipe/timer.c
@@ -84,6 +84,8 @@ static void ipipe_timer_default_release(struct ipipe_timer *timer)
 
 static int get_dev_mode(struct clock_event_device *evtdev)
 {
+
+if (evtdev != NULL) {
 	if (clockevent_state_oneshot(evtdev) ||
 		clockevent_state_oneshot_stopped(evtdev))
 		return CLOCK_EVT_MODE_ONESHOT;
@@ -95,6 +97,9 @@ static int get_dev_mode(struct clock_event_device *evtdev)
 		return CLOCK_EVT_MODE_SHUTDOWN;
 
 	return CLOCK_EVT_MODE_UNUSED;
+	}
+else
+	return CLOCK_EVT_MODE_UNUSED;
 }
 
 void ipipe_host_timer_register(struct clock_event_device *evtdev)
