diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index f19455a84563..c2958afef732 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -867,7 +867,8 @@ dtb-$(CONFIG_SOC_DRA7XX) += \
 	dra72-evm.dtb \
 	dra72-evm-revc.dtb \
 	dra71-evm.dtb \
-	dra76-evm.dtb
+	dra76-evm.dtb \
+	emac-som-5728m-350es.dtb
 dtb-merge-$(CONFIG_SOC_DRA7XX) += \
 	am57xx-evm.dtb \
 	am57xx-evm-reva3.dtb \
