diff --git a/driver/ddramc.c b/driver/ddramc.c
index e5e72a48..cf3ffbf4 100644
--- a/driver/ddramc.c
+++ b/driver/ddramc.c
@@ -426,6 +426,19 @@ static void ddram_reg_config(struct ddramc_register *ddramc_config)
 #error "LPDDR2 2Gb 32bit width with 4 banks is not supported!"
 #endif /* endif of CONFIG_LPDDR2_S4 */
 #endif /* endif of CONFIG_DBW_16 */
+
+#elif defined (CONFIG_DDR_4_GBIT)
+#if defined (CONFIG_DBW_16)
+#error "LPDDR2 4Gb 16bit width with  banks is not supported!"
+#else
+	/* 32 bit bus width */
+#if defined CONFIG_LPDDR2_S4
+	bank = AT91C_DDRC2_NB_BANKS_8;
+	row = AT91C_DDRC2_NR_14;
+	col = AT91C_DDRC2_NC_DDR10_SDR9;
+#endif /* endif of CONFIG_LPDDR2_S4 */
+#endif /* endif of CONFIG_DBW_16 */ 
+
 #endif /* endif of CONFIG_DDR_128_MBIT */
 #endif /* endif LPDDR2 */
 
