--- xmlrpc-c-1.39.07.orig/config.mk.in	2016-02-03 11:37:05.927214774 -0600
+++ xmlrpc-c-1.39.07/config.mk.in	2016-02-03 11:38:08.382434703 -0600
@@ -359,8 +359,8 @@
 # The cross compiling user can update config.mk or override
 # BUILDTOOL_CC on a make command.
 
-BUILDTOOL_CC = $(CC)
-BUILDTOOL_CCLD = $(CCLD)
+BUILDTOOL_CC = x86_64-linux-gnu-gcc
+BUILDTOOL_CCLD = x86_64-linux-gnu-gcc
 
 ifeq ($(C_COMPILER_GNU),yes)
   CFLAGS_NO_INLINE_WARNING = -Wno-inline
