diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index b9b8cbcfaf..5a3cc1fdf0 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -391,6 +391,7 @@ dtb-$(CONFIG_TARGET_AM57XX_EVM) += am57xx-beagle-x15.dtb \
 	am574x-idk.dtb \
 	am572x-idk.dtb	\
 	am571x-idk.dtb
+dtb-$(CONFIG_TARGET_SOM5728) += emac-som5728-350es.dtb
 dtb-$(CONFIG_TARGET_STV0991) += stv0991.dtb
 
 dtb-$(CONFIG_ARCH_LS1021A) += ls1021a-qds-duart.dtb \
diff --git a/arch/arm/mach-omap2/omap5/Kconfig b/arch/arm/mach-omap2/omap5/Kconfig
index a7132dae09..a01c1f9c56 100644
--- a/arch/arm/mach-omap2/omap5/Kconfig
+++ b/arch/arm/mach-omap2/omap5/Kconfig
@@ -41,6 +41,16 @@ config TARGET_AM57XX_EVM
 	imply SPL_THERMAL
 	imply TI_DRA7_THERMAL
 
+config TARGET_SOM5728
+	bool "EMACINC SOM5728"
+	select BOARD_LATE_INIT
+	select CMD_DDR3
+	select DRA7XX
+	imply DM_THERMAL
+	imply SCSI
+	imply SPL_THERMAL
+	imply TI_DRA7_THERMAL
+
 endchoice
 
 config SYS_SOC
@@ -159,5 +169,6 @@ endif
 source "board/ti/omap5_uevm/Kconfig"
 source "board/ti/dra7xx/Kconfig"
 source "board/ti/am57xx/Kconfig"
+source "board/emacinc/som5728/Kconfig"
 
 endif
