diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 965895bc2a..d2c5d34a04 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -884,7 +884,8 @@ dtb-$(CONFIG_MX6UL) += \
 	imx6ul-pico-hobbit.dtb \
 	imx6ul-pico-pi.dtb \
 	imx6ul-kontron-n631x-s.dtb \
-	imx6ull-kontron-n641x-s.dtb
+	imx6ull-kontron-n641x-s.dtb \
+	som-imx6ul-pmb.dts
 
 dtb-$(CONFIG_MX6ULL) += \
 	imx6ull-14x14-evk.dtb \
diff --git a/arch/arm/mach-imx/mx6/Kconfig b/arch/arm/mach-imx/mx6/Kconfig
index c7a03e5042..8122db54e7 100644
--- a/arch/arm/mach-imx/mx6/Kconfig
+++ b/arch/arm/mach-imx/mx6/Kconfig
@@ -440,6 +440,30 @@ config TARGET_MX6UL_14X14_EVK
 	select FSL_CAAM
 	select ARCH_MISC_INIT
 
+config TARGET_SOMIMX6UL_PMB
+	bool "Fuel Trax PMB boards"
+	depends on MX6UL
+	select BOARD_LATE_INIT
+	select DM
+	select SUPPORT_SPL
+	imply CMD_DM
+
+config TARGET_SOMIMX6UL_GSB
+	bool "Fuel Trax GSB boards"
+	depends on MX6UL
+	select BOARD_LATE_INIT
+	select DM
+	select SUPPORT_SPL
+	imply CMD_DM
+
+config TARGET_SOMIMX6UL
+        bool "EMAC IMX6UL boards"
+        depends on MX6UL 
+        select BOARD_LATE_INIT
+        select DM
+        select SUPPORT_SPL
+        imply CMD_DM
+	
 config TARGET_MX6UL_ENGICAM
 	bool "Support Engicam GEAM6UL/Is.IoT"
 	depends on MX6UL
@@ -489,6 +513,12 @@ config TARGET_MYS_6ULX
 	select DM_THERMAL
 	select SUPPORT_SPL
 
+config TARGET_SOMIMX6
+        bool "SoM-iMX6 boards"
+	depends on MX6QDL
+        select BOARD_LATE_INIT
+        select SUPPORT_SPL
+
 config TARGET_NITROGEN6X
 	bool "nitrogen6x"
 	depends on MX6DL || MX6Q || MX6QDL || MX6S
@@ -706,6 +736,10 @@ source "board/freescale/mx6sllevk/Kconfig"
 source "board/freescale/mx6sxsabresd/Kconfig"
 source "board/freescale/mx6sxsabreauto/Kconfig"
 source "board/freescale/mx6ul_14x14_evk/Kconfig"
+source "board/emac/somimx6ul-pmb/Kconfig"
+source "board/emac/somimx6ul-gsb/Kconfig"
+source "board/emac/somimx6/Kconfig"
+source "board/emac/somimx6ul/Kconfig"
 source "board/freescale/mx6ullevk/Kconfig"
 source "board/bosch/acc/Kconfig"
 source "board/grinn/liteboard/Kconfig"
