diff --git a/hw/xfree86/fbdevhw/fbdevhw.c b/hw/xfree86/fbdevhw/fbdevhw.c
index 3d8b92e66..8ba920b3e 100644
--- a/hw/xfree86/fbdevhw/fbdevhw.c
+++ b/hw/xfree86/fbdevhw/fbdevhw.c
@@ -151,7 +151,6 @@ xfree2fbdev_fblayout(ScrnInfoPtr pScrn, struct fb_var_screeninfo *var)
     var->xres_virtual = pScrn->displayWidth ? pScrn->displayWidth :
         pScrn->virtualX;
     var->yres_virtual = pScrn->virtualY;
-    var->bits_per_pixel = pScrn->bitsPerPixel;
     if (pScrn->defaultVisual == TrueColor ||
         pScrn->defaultVisual == DirectColor) {
         var->red.length = pScrn->weight.red;
@@ -163,6 +162,8 @@ xfree2fbdev_fblayout(ScrnInfoPtr pScrn, struct fb_var_screeninfo *var)
         var->green.length = 8;
         var->blue.length = 8;
     }
+
+    var->bits_per_pixel = var->red.length + var->green.length + var->blue.length;
 }
 
 static void
