diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 3e367f753f..ff32d15114 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1067,6 +1067,17 @@ dtb-$(CONFIG_TARGET_SAMA5D4_XPLAINED) += \
 dtb-$(CONFIG_TARGET_VINCO) += \
 	at91-vinco.dtb
 
+dtb-$(CONFIG_TARGET_IPAC9X25) += \
+	ipac-9x25.dtb
+
+dtb-$(CONFIG_TARGET_SOM9X25M) += \
+	som-9x25m.dtb
+
+dtb-$(CONFIG_TARGET_SOMA5D36) += \
+	som-a5d36.dtb \
+	som-a536mp.dtb 
+	
+
 dtb-$(CONFIG_ARCH_BCM283X) += \
 	bcm2835-rpi-a.dtb \
 	bcm2835-rpi-a-plus.dtb \
diff --git a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
index 6ac46d453c..afa24683bb 100644
--- a/arch/arm/mach-at91/Kconfig
+++ b/arch/arm/mach-at91/Kconfig
@@ -303,6 +303,27 @@ config TARGET_VINCO
 	select SUPPORT_SPL
 	imply CMD_DM
 
+config TARGET_IPAC9X25
+	bool "EMAC Inc. iPac-9x25 board."
+	select AT91SAM9X5
+	select BOARD_EARLY_INIT_F
+	select BOARD_LATE_INIT
+	select SUPPORT_SPL
+
+config TARGET_SOM9X25M
+	bool "EMAC Inc. SoM-9x25M System on Module"
+	select AT91SAM9X5
+	select BOARD_EARLY_INIT_F
+	select BOARD_LATE_INIT
+	select SUPPORT_SPL
+
+config TARGET_SOMA5D36
+	bool "EMAC Inc. SoM-A5D36 System on Module"
+	select BOARD_EARLY_INIT_F
+	select BOARD_LATE_INIT
+	select SAMA5D3
+	select SUPPORT_SPL
+
 endchoice
 
 config ATMEL_SFR
@@ -337,6 +358,9 @@ source "board/bluewater/gurnard/Kconfig"
 source "board/bluewater/snapper9260/Kconfig"
 source "board/calao/usb_a9263/Kconfig"
 source "board/egnite/ethernut5/Kconfig"
+source "board/emacinc/ipac9x25/Kconfig"
+source "board/emacinc/som9x25m/Kconfig"
+source "board/emacinc/soma5d36/Kconfig"
 source "board/esd/meesc/Kconfig"
 source "board/gardena/smart-gateway-at91sam/Kconfig"
 source "board/l+g/vinco/Kconfig"
