--- grub-2.02-old/Makefile.util.am	2017-04-24 06:30:20.000000000 -0500
+++ grub-2.02/Makefile.util.am	2018-09-28 15:33:32.431474765 -0500
@@ -677,7 +677,7 @@
 dist_noinst_DATA += util/grub.d/10_xnu.in 
 endif COND_HOST_XNU
 if COND_HOST_LINUX
-grubconf_SCRIPTS += 20_linux_xen
+grubconf_SCRIPTS +=
 
 20_linux_xen: $(top_builddir)/config.status util/grub.d/20_linux_xen.in 
 	(for x in util/grub.d/20_linux_xen.in ; do cat $(srcdir)/"$$x"; done) | $(top_builddir)/config.status --file=$@:-
@@ -687,7 +687,7 @@
 EXTRA_DIST += 
 dist_noinst_DATA += util/grub.d/20_linux_xen.in 
 endif COND_HOST_LINUX
-grubconf_SCRIPTS += 30_os-prober
+grubconf_SCRIPTS +=
 
 30_os-prober: $(top_builddir)/config.status util/grub.d/30_os-prober.in 
 	(for x in util/grub.d/30_os-prober.in ; do cat $(srcdir)/"$$x"; done) | $(top_builddir)/config.status --file=$@:-
--- grub-2.02-old/grub-core/boot/i386/pc/boot.S	2017-09-12 12:14:02.477510552 -0500
+++ grub-2.02/grub-core/boot/i386/pc/boot.S	2017-09-12 12:04:53.273754812 -0500
@@ -478,7 +478,7 @@
 LOCAL(stop):
 	jmp	LOCAL(stop)

-notification_string:	.asciz "GRUB "
+notification_string:	.asciz ""
 geometry_error_string:	.asciz "Geom"
 hd_probe_error_string:	.asciz "Hard Disk"
 read_error_string:	.asciz "Read"
--- grub-2.02-old/grub-core/boot/i386/pc/diskboot.S	2017-09-12 12:14:02.477510552 -0500
+++ grub-2.02/grub-core/boot/i386/pc/diskboot.S	2017-09-12 12:05:57.498662013 -0500
@@ -320,9 +320,9 @@
 /* go here when you need to stop the machine hard after an error condition */
 LOCAL(stop):	jmp	LOCAL(stop)

-notification_string:	.asciz "loading"
+notification_string:	.asciz ""

-notification_step:	.asciz "."
+notification_step:	.asciz ""
 notification_done:	.asciz "\r\n"

 geometry_error_string:	.asciz "Geom"
--- grub-2.02-old/grub-core/kern/main.c	2017-09-12 12:14:02.481510609 -0500
+++ grub-2.02/grub-core/kern/main.c	2017-09-12 12:07:51.204267993 -0500
@@ -270,9 +270,9 @@
   grub_boot_time ("After machine init.");

   /* Hello.  */
-  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
+  /*grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
   grub_printf ("Welcome to GRUB!\n\n");
-  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
+  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);*/

   grub_load_config ();

--- grub-2.02-old/grub-core/normal/menu.c	2017-09-12 12:14:02.481510609 -0500
+++ grub-2.02/grub-core/normal/menu.c	2017-09-12 12:08:50.829110062 -0500
@@ -342,7 +342,7 @@
 {
   int fallback_entry;

-  callback->notify_booting (entry, callback_data);
+  /*callback->notify_booting (entry, callback_data);*/

   grub_menu_execute_entry (entry, 1);
