diff --git a/Makefile.util.am b/Makefile.util.am
index 972b506..019b4c2 100644
--- a/Makefile.util.am
+++ b/Makefile.util.am
@@ -8052,7 +8052,7 @@ endif
 
 if COND_i386_pc
 if COND_HOST_LINUX
-grubconf_SCRIPTS += 20_linux_xen
+grubconf_SCRIPTS +=
 
 20_linux_xen: util/grub.d/20_linux_xen.in  $(top_builddir)/config.status
 	$(top_builddir)/config.status --file=$@:$<
@@ -8231,7 +8231,7 @@ dist_noinst_DATA += util/grub.d/30_os-prober.in
 endif
 
 if COND_i386_pc
-grubconf_SCRIPTS += 30_os-prober
+grubconf_SCRIPTS +=
 
 30_os-prober: util/grub.d/30_os-prober.in  $(top_builddir)/config.status
 	$(top_builddir)/config.status --file=$@:$<
diff --git a/grub-core/boot/i386/pc/boot.S b/grub-core/boot/i386/pc/boot.S
index 314f140..0b50d20 100644
--- a/grub-core/boot/i386/pc/boot.S
+++ b/grub-core/boot/i386/pc/boot.S
@@ -378,7 +378,7 @@ LOCAL(general_error):
 LOCAL(stop):
 	jmp	LOCAL(stop)
 
-notification_string:	.asciz "GRUB "
+notification_string:	.asciz ""
 geometry_error_string:	.asciz "Geom"
 hd_probe_error_string:	.asciz "Hard Disk"
 read_error_string:	.asciz "Read"
diff --git a/grub-core/boot/i386/pc/diskboot.S b/grub-core/boot/i386/pc/diskboot.S
index d030a14..34a2d41 100644
--- a/grub-core/boot/i386/pc/diskboot.S
+++ b/grub-core/boot/i386/pc/diskboot.S
@@ -320,9 +320,9 @@ LOCAL(general_error):
 /* go here when you need to stop the machine hard after an error condition */
 LOCAL(stop):	jmp	LOCAL(stop)
 
-notification_string:	.asciz "loading"
+notification_string:	.asciz ""
 
-notification_step:	.asciz "."
+notification_step:	.asciz ""
 notification_done:	.asciz "\r\n"
 
 geometry_error_string:	.asciz "Geom"
diff --git a/grub-core/kern/main.c b/grub-core/kern/main.c
index 3262444..4d892f0 100644
--- a/grub-core/kern/main.c
+++ b/grub-core/kern/main.c
@@ -211,9 +211,9 @@ grub_main (void)
   grub_machine_init ();
 
   /* Hello.  */
-  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
+/*  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
   grub_printf ("Welcome to GRUB!\n\n");
-  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
+  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);*/
 
   /* Load pre-loaded modules and free the space.  */
   grub_register_exported_symbols ();
diff --git a/grub-core/normal/menu.c b/grub-core/normal/menu.c
index 7e0a158..e01c7bd 100644
--- a/grub-core/normal/menu.c
+++ b/grub-core/normal/menu.c
@@ -287,7 +287,7 @@ grub_menu_execute_with_fallback (grub_menu_t menu,
 {
   int fallback_entry;
 
-  callback->notify_booting (entry, callback_data);
+  /*callback->notify_booting (entry, callback_data);*/
 
   grub_menu_execute_entry (entry, 1);
 
