diff --git a/Makefile.util.am b/Makefile.util.am
index 972b506..019b4c2 100644
--- a/Makefile.util.am
+++ b/Makefile.util.am
@@ -8052,7 +8052,7 @@ endif
 
 if COND_i386_pc
 if COND_HOST_LINUX
-grubconf_SCRIPTS += 20_linux_xen
+grubconf_SCRIPTS +=
 
 20_linux_xen: util/grub.d/20_linux_xen.in  $(top_builddir)/config.status
 	$(top_builddir)/config.status --file=$@:$<
@@ -8231,7 +8231,7 @@ dist_noinst_DATA += util/grub.d/30_os-prober.in
 endif
 
 if COND_i386_pc
-grubconf_SCRIPTS += 30_os-prober
+grubconf_SCRIPTS +=
 
 30_os-prober: util/grub.d/30_os-prober.in  $(top_builddir)/config.status
 	$(top_builddir)/config.status --file=$@:$<
