diff --git a/board/at91sam9x5ek/at91sam9x5ek.c b/board/at91sam9x5ek/at91sam9x5ek.c
index f87d80d..3bfe47e 100644
--- a/board/at91sam9x5ek/at91sam9x5ek.c
+++ b/board/at91sam9x5ek/at91sam9x5ek.c
@@ -70,13 +70,21 @@ static void ddramc_reg_config(struct ddramc_register *ddramc_config)
 	ddramc_config->mdr = (AT91C_DDRC2_DBW_16_BITS
 			| AT91C_DDRC2_MD_DDR2_SDRAM);
 
+#ifdef CONFIG_RAM_64MB
+	ddramc_config->cr = (AT91C_DDRC2_NC_DDR10_SDR9 /* 10 column bits(1K) */
+			| AT91C_DDRC2_NR_13              /* 13 row bits (8K) */
+			| AT91C_DDRC2_CAS_3              /* CAS Latency 3 */
+			| AT91C_DDRC2_NB_BANKS_4         /* 4 banks */
+			| AT91C_DDRC2_DLL_RESET_DISABLED /* DLL not reset */
+			| AT91C_DDRC2_DECOD_INTERLEAVED);/*Interleaved decode*/
+#else
 	ddramc_config->cr = (AT91C_DDRC2_NC_DDR10_SDR9 /* 10 column bits(1K) */
 			| AT91C_DDRC2_NR_13              /* 13 row bits (8K) */
 			| AT91C_DDRC2_CAS_3              /* CAS Latency 3 */
 			| AT91C_DDRC2_NB_BANKS_8         /* 8 banks */
 			| AT91C_DDRC2_DLL_RESET_DISABLED /* DLL not reset */
 			| AT91C_DDRC2_DECOD_INTERLEAVED);/*Interleaved decode*/
-
+#endif
 	/*
 	 * Make sure to uncomment the following line if the DDR controller
 	 * shares the EBI with another memory controller (SMC, NAND,..).
