diff --git a/board/at91sam9m10g45ek/at91sam9m10g45ek.c b/board/at91sam9m10g45ek/at91sam9m10g45ek.c
index 6402dac..1ba57a4 100644
--- a/board/at91sam9m10g45ek/at91sam9m10g45ek.c
+++ b/board/at91sam9m10g45ek/at91sam9m10g45ek.c
@@ -160,6 +160,17 @@ static int ek_special_hw_init(void)
 	 return 0;
 }
 
+static void tri_pb18(void)
+{
+       /* Configure PB18 CAN Rx to input no pull-up */
+       const struct pio_desc wdt_pins[] = {
+               {"WDT", AT91C_PIN_PB(18), 0, PIO_DEFAULT, PIO_PERIPH_A},
+               {(char *)0, 0, 0, PIO_DEFAULT, PIO_PERIPH_A},
+       };
+
+       pio_configure(wdt_pins);
+}
+
 #ifdef CONFIG_HW_INIT
 void hw_init(void)
 {
@@ -201,6 +212,9 @@ void hw_init(void)
 #endif
 	/* do some special init */
 	ek_special_hw_init();
+
+	/* tristate pb18 - used as watchdog input on some carriers */
+	tri_pb18();
 }
 #endif /* #ifdef CONFIG_HW_INIT */
 
