diff --git a/kernel/ipipe/timer.c b/kernel/ipipe/timer.c
index 22d5e5be874a..9920d878ef6e 100644
--- a/kernel/ipipe/timer.c
+++ b/kernel/ipipe/timer.c
@@ -83,17 +83,19 @@ static void ipipe_timer_default_release(struct ipipe_timer *timer)
 
 static int get_dev_mode(struct clock_event_device *evtdev)
 {
-	if (clockevent_state_oneshot(evtdev) ||
-		clockevent_state_oneshot_stopped(evtdev))
-		return CLOCK_EVT_MODE_ONESHOT;
-
-	if (clockevent_state_periodic(evtdev))
-		return CLOCK_EVT_MODE_PERIODIC;
+	if (evtdev != NULL) {
+		if (clockevent_state_oneshot(evtdev) ||
+			clockevent_state_oneshot_stopped(evtdev))
+			return CLOCK_EVT_MODE_ONESHOT;
 
-	if (clockevent_state_shutdown(evtdev))
-		return CLOCK_EVT_MODE_SHUTDOWN;
+		if (clockevent_state_periodic(evtdev))
+			return CLOCK_EVT_MODE_PERIODIC;
 
-	return CLOCK_EVT_MODE_UNUSED;
+		if (clockevent_state_shutdown(evtdev))
+			return CLOCK_EVT_MODE_SHUTDOWN;
+		return CLOCK_EVT_MODE_UNUSED;
+	} else
+		return CLOCK_EVT_MODE_UNUSED;
 }
 
 void ipipe_host_timer_register(struct clock_event_device *evtdev)
